slyd/slyd/gitstorage/projects.py

```diff
@@ -181,8 +198,12 @@ def copy_data(self, source, destination, spiders, items):
         copier = GitSpiderCopier(source, destination, branch)
         return json.dumps(copier.copy(spiders, items))
 
-    def download_project(self, name, spiders=None, version=None, fmt=None):
-        self._open_repo(name, read_only=True)
+    def download_project(self, name, spiders=None, version=None, fmt=None,
+                         branch=None, **kwargs):
```

Comment:
Here's the spec for downloading from the FS storage implementation
https://github.com/scrapinghub/portia/pull/536/files#diff-5f33b932b8cf1d9581ad6fefd4f676f3R237
Here's where the download gets called:
https://github.com/scrapinghub/portia/pull/536/files#diff-909c4e43c52fe1edc233a4b28ff41609R97

The Git backend takes optional `branch` and `version` arguments but the FS backend does not. If we were to add an arg to the FS backend that is not used by the Git backend and there was no kwargs there then it would throw a TypeError because it got an unexpected additional argument.

So to answer your question it is not used but is there for consistency between the backends.